daemon/cmd/daemon.go

```diff
@@ -395,6 +387,20 @@ func NewDaemon(ctx context.Context, dp datapath.Datapath) (*Daemon, *endpointRes
 	treatRemoteNodeAsHost := option.Config.AlwaysAllowLocalhost() && !option.Config.EnableRemoteNodeIdentity
 	policyApi.InitEntities(option.Config.ClusterName, treatRemoteNodeAsHost)
 
+	d.k8sCachesSynced = d.k8sWatcher.InitK8sSubsystem()
```

Comment:
Not sure if I'm missing something here, but shouldn't we guard this statement by checking if K8s is enabled? What would happen if K8s is disabled?

Reply:
It was not protected before which is why I've kept it this way. The check is embedded in `EnableK8sWatcher()`.